src/apps/api/tests/test_users.py

```diff
@@ -0,0 +1,110 @@
+        user = user or self.user
+        email_pk = EmailAddress.objects.get(email=email_address).id
+        return self.client.post(
+            reverse('user-change_primary_email', kwargs={'pk': user.id, 'version': 'v1'}),
+            data={'email_pk': email_pk}
+        )
+
+    def test_add_email_permissions(self):
+        email1, email2, email3 = ('email1@example.com', 'email2@example.com', 'email3@example.com')
+        emails = self.user.email_addresses.all().values_list('email', flat=True)
+        assert email1 not in emails and email2 not in emails and email3 not in emails
+        self.login(username='test')
+        resp = self.add_email(email1)
+        assert resp.status_code == 201
+        self.login(username='admin')
+        resp = self.add_email(email2)
+        assert resp.status_code == 201
+        self.login(username='norm')
+        resp = self.add_email(email3)
+        assert resp.status_code == 403
+        emails = self.user.email_addresses.all().values_list('email', flat=True)
+        assert email1 in emails
+        assert email2 in emails
+        assert email3 not in emails
+
+    def test_remove_email_permissions(self):
+        email1, email2, email3 = ('email1@example.com', 'email2@example.com', 'email3@example.com')
+        for email in [email1, email2, email3]:
+            EmailAddressFactory(user=self.user, email=email)
+        emails = self.user.email_addresses.all().values_list('email', flat=True)
+        assert email1 in emails and email2 in emails and email3 in emails
+        self.login(username='test')
+        resp = self.remove_email(email1)
+        assert resp.status_code == 200
+        self.login(username='admin')
+        resp = self.remove_email(email2)
+        assert resp.status_code == 200
+        self.login(username='norm')
+        resp = self.remove_email(email3)
+        assert resp.status_code == 403
+        emails = self.user.email_addresses.all().values_list('email', flat=True)
+        assert email1 not in emails
+        assert email2 not in emails
+        assert email3 in emails
+
+    def test_change_primary_email_permissions(self):
+        email1, email2, email3 = ('email1@example.com', 'email2@example.com', 'email3@example.com')
+        for email in [email1, email2, email3]:
+            EmailAddressFactory(user=self.user, email=email, verified=True)
+        assert self.user.primary_email not in [email1, email2, email3]
+        self.login(username='test')
+        resp = self.change_primary(email1)
+        assert resp.status_code == 200
+        assert self.user.primary_email == email1
+        self.login(username='admin')
+        resp = self.change_primary(email2)
+        assert resp.status_code == 200
+        assert self.user.primary_email == email2
+        self.login(username='norm')
+        resp = self.change_primary(email3)
+        assert resp.status_code == 403
+        assert self.user.primary_email == email2  # didn't change
```

Comment:
no, because `primary_email` is a property that is recalculated per call, and it queries `EmailAddress` objects, so no relevant data has changed on the user model. (evidenced by the fact that both lines 89 and 93 pass without refetching the user)